src/components/plots/styling/frequency/SelectCellSets.jsx

```diff
@@ -29,31 +29,27 @@ const SelectCellSets = (props) => {
     onUpdate, config,
   } = props;
 
-  const optionsMetadata = useSelector(getCellSetsHierarchyByType('metadataCategorical'));
-  const optionsCellSets = useSelector(getCellSetsHierarchyByType('cellSets'));
-
-  let disabled = false;
-  let toolTipText;
-
   const changeXAxisGrouping = (option) => {
-    onUpdate({ xAxisGrouping: option.value });
+    onUpdate({
+      xAxisGrouping: option.value,
+      axes: { ...config.axes, xAxisText: option.label },
+    });
   };
 
   const changeProportionGrouping = (option) => {
     onUpdate({ proportionGrouping: option.value });
   };
 
-  const metadataMenu = getSelectOptions(optionsMetadata);
-  const cellSetMenu = getSelectOptions(optionsCellSets);
-
-  let menuValue;
+  const metadataTypes = useSelector(getCellSetsHierarchyByType('metadataCategorical'));
+  const cellSetsTypes = useSelector(getCellSetsHierarchyByType('cellSets'));
+  const metadataGroupByOptions = getSelectOptions(metadataTypes);
+  const cellSetsGroupByOptions = getSelectOptions(cellSetsTypes);
 
-  if (!metadataMenu) {
-    menuValue = 'sample';
-    disabled = true;
-    toolTipText = 'The x-axis cannot be changed as this dataset has only a single sample.';
-  } else {
-    menuValue = config.xAxisGrouping;
+  let metadataToGroupByDisabled = false;
+  let tooltipText;
+  if (!metadataGroupByOptions || metadataGroupByOptions.length === 1) {
+    metadataToGroupByDisabled = true;
+    tooltipText = 'The x-axis cannot be changed as this dataset has only a single sample.';
```

Comment:
I found out that this code has a bug -- the tooltip with the message was never displayed, even in cases when we have only 1 sample. So I fixed it.